www/content/tutorial.md

```diff
@@ -1586,154 +1570,276 @@ app "cli-tutorial"
 
--   `await` is not a language keyword in Roc! It's referring to the `Task.await` function, which we imported unqualified by writing `Task.{ await }` in our module imports. (That said, it is playing a similar role here to the `await` keyword in languages that have `async`/`await` keywords, even though in this case it's a function instead of a special keyword.)
--   Backpassing syntax does not need to be used with `await` in particular. It can be used with any function.
--   Roc's compiler treats functions defined with backpassing exactly the same way as functions defined the other way. The only difference between `\input ->` and `input <-` is how they look, so feel free to use whichever looks nicer to you!
+Note that the `!` moves when we do this. It's no longer after `Stdout.line`, but rather after `Task.mapErr` instead:
+
+```roc
+    Stdout.line "Type something and press Enter."
+    |> Task.mapErr! PrintPrompt
+```
+
+This code is doing three things:
+1. Call `Stdout.line "..."`, which returns a `Task` value
+2. Transform that `Task` value into another `Task` value using `|> Task.mapErr`
+3. Wait until that final `Task` value (returned by `mapErr`) successfully completes, using `!`
+
+It's easier to see why the `!` needs to move when considering a `Task` that produces a useful value, like `Stdin.line`. Compare these two:
+
+```roc
+    input =
+        Stdin.line
+        |> Task.mapErr! ReadStdin
+```
+
+This versions starts with the `Stdin.line` task, then passes it to `Task.mapErr` to tag its error with `ReadStdin`, and then finally uses `!` to await that final transformed task that `Task.mapErr` returned.
+
+```roc
+    input =
+        Stdin.line!
+        |> Task.mapErr ReadStdin
+```
+
+This version starts with the `Stdin.line` task, awaits it with `!` to get its `Str` value if it succeded, and then tries to call `Task.mapErr` on that `Str` value. This will be a type mismatch!
+
+In general, an easy rule to remember is "do the `!` last" - if you want to modify a `Task` before running it, use the `!` only after you've made all the modifications and ended up with the final `Task` you'd like to run.
 
 See the [Task & Error Handling example](https://www.roc-lang.org/examples/Tasks/README.html) for a more detailed explanation of how to use tasks to help with error handling in a larger program.
 
+### [Displaying Roc values with `Inspect.toStr`](#inspect) {#inspect}
+
+The [`Inspect.toStr`](https://www.roc-lang.org/builtins/Inspect#toStr) function returns a `Str` representation of any Roc value. It's useful for things like debugging and logging (although [`dbg`](https://www.roc-lang.org/tutorial#debugging) is often nicer for debugging in particular), but its output is almost never something that should be shown to end users! In this case we're just using it for our own learning, but it would be better to run a `when` on `e` and display a more helpful message.
```

Comment:
Might be worthwhile linking to https://www.roc-lang.org/abilities#inspect-ability

Also jus noticed that has been merged, and now also needs to be updated. Issue added https://github.com/roc-lang/roc/issues/6679